libbpf-rs/src/program.rs

```diff
@@ -52,7 +53,7 @@ pub struct UprobeOpts {
     /// `func_name` and use `func_offset` argument to specify offset within the
     /// function. Shared library functions must specify the shared library
     /// binary_path.
-    pub func_name: String,
+    pub func_name: Option<String>,
```

Comment:
It seems the documentation would need to be updated accordingly. Also, `func_offset` appears to be a bit of a misnomer now. Should probably be renamed to `offset`.

Reply:
Just added a test. As for the naming and the comments, it seems all derive from libbpf.